src/main/java/gyro/aws/ec2/ConnectionNotificationResource.java

```diff
@@ -167,13 +168,21 @@ public void setType(String type) {
     @Override
     public void copyFrom(ConnectionNotification connectionNotification) {
         setId(connectionNotification.connectionNotificationId());
-        setConnectionEvents(connectionNotification.connectionEvents() != null ? new HashSet<>(connectionNotification.connectionEvents()) : null);
-        setArn(connectionNotification.connectionNotificationArn());
+        setConnectionEvents(connectionNotification.connectionEvents() != null
+            ? new HashSet<>(connectionNotification.connectionEvents())
+            : null);
+        setTopic(findById(TopicResource.class, connectionNotification.connectionNotificationId()) != null ? findById(
+                TopicResource.class,
+                connectionNotification.connectionNotificationId()) : null);
```

Comment:
Topic is a required field, so will always be present
```suggestion
        setTopic(findById(TopicResource.class, connectionNotification.connectionNotificationId()));
```